lua/luasnip/extras/treesitter_postfix.lua

```diff
@@ -0,0 +1,320 @@
+			captures = captures,
+			clear_region = {
+				from = {
+					start_row,
+					start_col,
+				},
+				to = {
+					cursor[1] - 1,
+					cursor[2],
+				},
+			},
+			env_override = {
+				TREESITTER_MATCHES = matches,
+				TREESITTER_BEST_MATCH = {
+					start = { start_col, start_col },
+					text = best_match_text,
+					type = match_result.best_match:type(),
+				},
+			},
+		}
+
+		if user_resolver then
+			local user_res = user_resolver(
+				snippet,
+				line_to_cursor,
+				matched_trigger,
+				captures
+			)
+			if user_res then
+				ret = vim.tbl_deep_extend(
+					"force",
+					ret,
+					user_res,
+					{ env_override = {} }
+				)
+			else
+				return nil
+			end
+		end
+
+		return ret
+	end
+end
+
+local function generate_simple_parent_lookup_function(lookup_fun)
+	---@param types string|string[]
+	---@param trigger string
+	---@return LuaSnip.extra.MatchTSNodeFunc
+	return function(types, trigger)
+		local type_checker = tbl.normalize_search_table(types)
+		---@param parser LuaSnip.extra.TSParser
+		---@param pos { [1]: number, [2]: number }?
+		return function(parser, pos)
+			local root = parser:get_node_at_pos(pos)
+
+			if root == nil then
+				return
+			end
+
+			local exclude_pos = util.get_cursor_0ind()
+			if type(parser.source) == "string" then
+				-- buffer has re-parsed
+				exclude_pos[2] = exclude_pos[2] - #trigger - 1
+			else
+				exclude_pos[2] = exclude_pos[2] - 1
+			end
+
+			---@param node TSNode
+			local check_node_exclude_pos = function(node)
+				return not vim.treesitter.is_in_node_range(
+					node,
+					exclude_pos[1],
+					exclude_pos[2]
+				)
+			end
+			---@param node TSNode
+			local check_node_type = function(node)
+				return type_checker[node:type()]
+			end
+
+			return lookup_fun(root, function(node)
+				return check_node_type(node) and check_node_exclude_pos(node)
+			end)
+		end
+	end
+end
+
+local find_topmost_types =
+	generate_simple_parent_lookup_function(ts.find_topmost_parent)
+local find_first_types =
+	generate_simple_parent_lookup_function(ts.find_first_parent)
+
+---@param n number
+local function find_nth_parent(n)
+	---@param parser LuaSnip.extra.TSParser
+	---@param pos { [1]: number, [2]: number }?
+	return function(parser, pos)
+		local root = parser:get_node_at_pos(pos)
+
+		if root == nil then
+			return
+		end
+
+		---@param node TSNode
+		local check_node_exclude_pos = function(node)
+			local exclude_pos = util.get_cursor_0ind()
+			exclude_pos[2] = exclude_pos[2] - 1
+			return not vim.treesitter.is_in_node_range(
+				node,
+				exclude_pos[1],
+				exclude_pos[2]
+			)
+		end
+		return ts.find_nth_parent(root, n, check_node_exclude_pos)
+	end
+end
+
+local function treesitter_postfix(context, nodes, opts)
+	opts = opts or {}
+	vim.validate({
+		context = { context, { "string", "table" } },
+		nodes = { nodes, "table" },
+		opts = { opts, "table" },
+	})
+
+	context = node_util.wrap_context(context)
+	context.wordTrig = false
+	context.trigEngine = "plain"
```

Comment:
I don't see why this has to be plain, could you enlighten me?

Reply:
Emmm, it just copy it from my config. This can be deleted.